ang/civicase/Dashboard.js

```diff
@@ -0,0 +1,207 @@
+        'activity_type_id.grouping': {LIKE: '%communication%'},
+        'status_id.filter': 1,
+        is_current_revision: 1,
+        is_test: 0
+      }, $scope.activityFilters)]);
+      // next milestones
+      apiCalls.push(['Activity', 'get', _.extend({
+        'activity_type_id.grouping': {LIKE: '%milestone%'},
+        'status_id.filter': 0,
+        options: {limit: activitiesToShow, sort: 'activity_date_time ASC'}
+      }, params)]);
+      apiCalls.push(['Activity', 'getcount', _.extend({
+        'activity_type_id.grouping': {LIKE: '%milestone%'},
+        'status_id.filter': 0,
+        is_current_revision: 1,
+        is_test: 0
+      }, $scope.activityFilters)]);
+      crmApi(apiCalls).then(function (data) {
+        $scope.$broadcast('caseRefresh');
+        $scope.summaryData = data[apiCalls.length - 5].values;
+        $scope.dashboardActivities.recentCommunication = _.each(data[apiCalls.length - 4].values, formatActivity);
+        $scope.dashboardActivities.recentCommunicationCount = data[apiCalls.length - 3];
+        $scope.dashboardActivities.nextMilestones = _.each(data[apiCalls.length - 2].values, formatActivity);
+        $scope.dashboardActivities.nextMilestonesCount = data[apiCalls.length - 1];
+      });
+    };
+
+    /**
+     * Bind route paramaters to scope variables
+     */
+    function bindRouteParamsToScope () {
+      $scope.$bindToRoute({ param: 'dtab', expr: 'activeTab', format: 'int', default: 0 });
+      $scope.$bindToRoute({ param: 'dbd', expr: 'showBreakdown', format: 'bool', default: false });
+      $scope.$bindToRoute({ param: 'drel', expr: 'filters.caseRelationshipType', format: 'raw', default: 'is_involved' });
+    }
+
+    /**
+     * Watcher for caseRelationshipType
+     *
+     * @param {String} newValue
+     */
+    function caseRelationshipTypeWatcher (newValue) {
+      newValue === 'is_case_manager'
+        ? $scope.activityFilters.case_filter.case_manager = [CRM.config.user_contact_id]
+        : delete ($scope.activityFilters.case_filter.case_manager);
+
+      newValue === 'is_involved'
+        ? $scope.activityFilters.case_filter.contact_id = [CRM.config.user_contact_id]
+        : delete ($scope.activityFilters.case_filter.contact_id);
+    }
+
+    /**
+     * Prepate case filter options
+     */
+    function prepareCaseFilterOption () {
+      var options = [
+        { 'text': 'My cases', 'id': 'is_case_manager' },
+        { 'text': 'Cases I am involved in', 'id': 'is_involved' }
+      ];
+
+      if (CRM.checkPerm('access all cases and activities')) {
+        options.push({ 'text': 'All Cases', 'id': 'all' });
+      }
+
+      $scope.caseRelationshipOptions = options;
+    }
+
+    /**
+     * Initialise watchers
+     */
+    function initWatchers () {
+      $scope.$watch('filters.caseRelationshipType', caseRelationshipTypeWatcher);
+    }
+
+    // Translate between the dashboard's global filter-options and
```

Comment:
Lets remove it if we don't need this piece of code.

Reply:
It kept as it might be required in the Dashboard tab, added a TODO though.

Reply:
Yes that is fine too. Thanks